ActivatePubKey/handler.ts

```diff
@@ -0,0 +1,200 @@
+import { JwkPublicKeyFromToken } from "@pagopa/ts-commons/lib/jwk";
+import { ActivatedPubKey } from "../generated/definitions/internal/ActivatedPubKey";
+import { AssertionRef } from "../generated/definitions/internal/AssertionRef";
+import { ActivatePubKeyPayload } from "../generated/definitions/internal/ActivatePubKeyPayload";
+import { AssertionWriter, PopDocumentWriter } from "../utils/writers";
+import { PopDocumentReader } from "../utils/readers";
+import { JwkPubKeyHashAlgorithmEnum } from "../generated/definitions/internal/JwkPubKeyHashAlgorithm";
+import {
+  AssertionFileName,
+  PendingLolliPopPubKeys,
+  TTL_VALUE_AFTER_UPDATE
+} from "../model/lollipop_keys";
+import { PubKeyStatusEnum } from "../generated/definitions/internal/PubKeyStatus";
+import {
+  retrievedLollipopKeysToApiActivatedPubKey,
+  retrievedValidPopDocument
+} from "../utils/lollipopKeys";
+import {
+  getAlgoFromAssertionRef,
+  getAllAssertionsRef
+} from "../utils/lollipopKeys";
+import { domainErrorToResponseError } from "../utils/errors";
+
+type ActivatePubKeyHandler = (
+  context: Context,
+  assertion_ref: AssertionRef,
+  body: ActivatePubKeyPayload
+) => Promise<
+  | IResponseSuccessJson<ActivatedPubKey>
+  | IResponseErrorNotFound
+  | IResponseErrorValidation
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+>;
+
+export const ActivatePubKeyHandler = (
+  popDocumentReader: PopDocumentReader,
+  popDocumentWriter: PopDocumentWriter,
+  assertionWriter: AssertionWriter
+): ActivatePubKeyHandler => (
+  _,
+  assertion_ref,
+  body
+): ReturnType<ActivatePubKeyHandler> =>
+  pipe(
+    popDocumentReader(assertion_ref),
+    TE.mapLeft(domainErrorToResponseError),
+    TE.filterOrElseW(PendingLolliPopPubKeys.is, () =>
+      ResponseErrorInternal("Unexpected status on pop document")
+    ),
+    TE.chainW(popDocument =>
+      pipe(
+        `${body.fiscal_code}-${assertion_ref}`,
+        AssertionFileName.decode,
+        TE.fromEither,
+        TE.bindTo("assertionFileName"),
+        TE.mapLeft(errors =>
+          ResponseErrorInternal(
+            `Could not decode assertionFileName | ${readableReport(errors)}`
+          )
+        ),
+        TE.chainFirst(({ assertionFileName }) =>
+          pipe(
+            assertionWriter(assertionFileName, body.assertion),
+            TE.mapLeft(error => ResponseErrorInternal(error.detail))
+          )
+        ),
+        TE.bind("jwkPubKeyFromString", () =>
+          pipe(
+            popDocument.pubKey,
+            JwkPublicKeyFromToken.decode,
+            TE.fromEither,
+            TE.mapLeft(errors =>
+              ResponseErrorInternal(
+                `Could not decode public key | ${readableReport(errors)}`
+              )
+            )
+          )
+        ),
+        TE.bind("assertionRefs", ({ jwkPubKeyFromString }) =>
+          pipe(
+            getAllAssertionsRef(
+              JwkPubKeyHashAlgorithmEnum.sha512,
```

Comment:
use MASTER_HASH_ALGO const

Reply:
Done in [d450057](https://github.com/pagopa/io-functions-lollipop/pull/13/commits/d4500570e87494922f1c25cc1197a1a58287cc75)